academic_observatory_workflows/workflows/openalex_telescope.py

```diff
@@ -300,13 +371,66 @@ def __init__(
 
         self.add_setup_task(self.check_dependencies)
         self.add_task(self.write_transfer_manifest)
-        # self.add_task(self.upload_transfer_manifest) #TODO uncomment when using transfer manifest
+        # self.add_task(self.upload_transfer_manifest)
         self.add_task(self.transfer)
         self.add_task(self.download_transferred)
         self.add_task(self.transform)
         self.add_task(self.upload_transformed)
-        self.add_task(self.bq_load_partition)
-        self.add_task_chain([self.bq_delete_old, self.bq_append_new, self.cleanup, self.add_new_dataset_releases])
+        self.add_task(self.bq_append_new)
+        self.add_task(self.bq_delete_old)
+        self.add_task(self.bq_create_snapshot)
+        self.add_task(self.add_new_dataset_releases)
+
+    def get_release_info(self, **kwargs) -> Tuple[pendulum.DateTime, pendulum.DateTime, bool]:
```

Comment:
I feel like we should be trying to use the get_release_info from the StreamTelescope rather than making a whole new method if possible. 

For OpenAlex, there is a top level DatasetType called OpenAlex which should have the correct end date set. This line `datasets = get_datasets(workflow_id=self.workflow_id)` in StreamTelescope would need to filter the datasets by a dataset_type.type_id, as there can be more than one dataset type. The type_id would need to be set in the constructor of each workflow.

Then `start_date = pendulum.instance(latest_release.created).start_of("day")` can be set back to `start_date = pendulum.instance(latest_release.end_date).start_of("day")`.

Reply:
So you mean I would update this method for the stream telescope to always use type_id 0? (or 1, not sure whether we're starting at 0 or 1)

Reply:
Ah never mind, the dataset_type.type_id is actually a string I see now.